src/pages/HomePage/HomePage.tsx

```diff
@@ -1,18 +1,66 @@
 import React from 'react';
-// import { Button } from '../../components/Button/Button';
+import useUserInfo from '../../hooks/useUserInfo';
+import { Button } from '../../components/Button/Button';
 import { Header } from '../../components/Header/Header';
 import { Menu } from '../../components/Menu/Menu';
 import { ContainerMain } from '../../components/ContainerMain/ContainerMain';
 import { Submenu } from '../../components/Submenu/Submenu';
+import {
+  GoogleLogin,
+  GoogleLoginResponse,
+  GoogleLoginResponseOffline,
+  GoogleLogout,
+} from 'react-google-login';
+import { UserProfile } from '../../models/UserProfile';
+
+export interface LoginState {
+  user: UserProfile | null;
+}
+
+const initialState: LoginState = { user: null };
+
+export const CurrentUserContext = React.createContext<LoginState>(initialState);
```

Comment:
`CurrentUserContext` - would be better to name it as `LoginContext` or somethig else. You should store other loging information here